### ISSUE TITLE
Bus error with conda-build 2.0 in Constructor installation

### ISSUE THREAD

**Issue by sscherfke** [conda/conda-build#1551]
We are building our own Conda distribution (similar to miniconda) using [constructor](https://github.com/conda/constructor/).

When we build *Python* (3.5.2 or 3.6.0b4) with *conda-build 1.21.11*, everything works fine and as expected.  When we build it with *conda-build 2.0.x*, the Conda installer generated by Constructor fails with:

```
./emsconda.sh: Zeile 161: 23713 Bus error $PYTHON -E -s $PREFIX/pkgs/.install.py $INST_OPT
```

Context from the installer:

```bash
install_dist()
{
    # This function installs a conda package into prefix, but without linking
    # the conda packages.  It untars the package and calls a simple script
    # which does the post extract steps (update prefix files, run 'post-link',
    # and creates the conda metadata).  Note that this is all done without
    # conda.
    echo "installing: $1 ..."
    PKG=$PREFIX/pkgs/$1.tar.bz2
    tar xjf $PKG -C $PREFIX --no-same-owner || exit 1
    if [[ $1 == 'python-3.5.2-6' ]]; then
        $PYTHON -E -V
        if (( $? )); then
            echo "ERROR:
cannot execute native linux-64 binary, output from 'uname -a' is:" >&2
            uname -a
            exit 1
        fi
    fi
    $PYTHON -E -s $PREFIX/pkgs/.install.py $INST_OPT || exit 1
    rm $PKG
}

install_dist python-3.5.2-6
```

The error happens when `.install.py:update_prefix()` tries to open `lib/libpython3.5m.so.1.0` for writing (obviously, Python does not not like its lib being edited while it is running).

The reason why it works with conda-build 1.21 but not with 2.0 is that 1.21 inserts prefixes only in the executable files, but not in the `*.so` files.  2.0 also inserts prefixes in `*.so` which the installer then tries to fix (and then fails).

Output of `conda info`:
```
Current conda install:

               platform : linux-64
          conda version : 4.2.12
       conda is private : False
      conda-env version : 2.5.1
    conda-build version : 2.0.0 (or 1.21.11 in the working case)
         python version : 3.5.2.final.0
       requests version : 2.11.1
       root environment : /tmp/py3  (writable)
    default environment : /tmp/py3
       envs directories : /tmp/py3/envs
          package cache : /tmp/py3/pkgs
           channel URLs : file:///tmp/py3/conda-bld/linux-64
                          file:///tmp/py3/conda-bld/noarch
            config file : /tmp/py3/.condarc
           offline mode : False
```

I can provide the installers (a working one and a failing one) and also the recipes if necessary.

**Comment by msarahan** [comment]
Thanks, but I disagree a bit with your diagnosis.  I think that the
difference between conda-build 1 and 2 that is affecting this is that
conda-build 1 was opt-in for prefix replacement, whereas conda-build 2 is
opt-out.

The docs reflect this, though they should be clearer about the different
behavior:

http://conda.pydata.org/docs/building/meta-yaml.html#binary-relocation

ignore_prefix_files in lib/libpython3.5m.so.1.0 seems like it might work?

http://conda.pydata.org/docs/building/meta-yaml.html#ignore-prefix-files

Let me know if these bits help you sort it out.  Also, conda-build 2.0.0 is
pretty old.  There are lots of bug fixes between that and the current
release (2.0.10) - please update before investigating further.

**Comment by sscherfke** [comment]
Thanks for your feedback. 

> ignore_prefix_files in lib/libpython3.5m.so.1.0 seems like it might work?

Adding `lib/libpython3.5m.so.1.0` and all `lib/python3.5/lib-dynload/*.so` to `build:ignore_prefix_files` helped indeed. It would be cool though if we could use Wildcards there so that we don't have to list every single `*.so` in `lib-dynload`.

I wonder why the [conda forge Python](https://github.com/conda-forge/python-feedstock/blob/master/recipe/meta.yaml) works without `ignore_prefix_files`?

> Also, conda-build 2.0.0 is pretty old. There are lots of bug fixes between that and the current release (2.0.10) - please update before investigating further.

We are using the lastest versions, of course. I just wanted to narrow down the issue to the last version were it worked and the earliest version were it stopped working. ;-)

Thanks for your help, again! :)

**Comment by msarahan** [comment]
I'm guessing that conda-forge is still using conda-build 1.

I'll look into adding globs for ignore_prefix_files soon.  That's a good
idea.
